### DISCUSSION TITLE
Not sure I understand the tag policies

### DISCUSSION THREAD

**Discussion by MikeHingley22** [dotnet/dotnet-docker#4971] [Q&A]
So - here's my understanding : 

if I have a docker file that starts : 

FROM mcr.microsoft.com/dotnet/sdk:6.0 as build 

Then it will load the latest SDK:6.0 tagged image appropriate for my environment - so it should pick 6.0.24  - because the floating version tag fixes the major and minor - but not the patch level.

I scanned my container that I built - and it contained a vulnerability [CVE-2023-35391](https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-35391) - which was fixed in 6.0.21 - which was released in August? 

By my reckoning the CVE shouldn't be there?

**Comment by MichaelSimons** [comment]
What version do you see emitted when you run `docker run --rm mcr.microsoft.com/dotnet/sdk:6.0 dotnet --info`?  The reason I ask is that docker caches images.  If you had previously pulled the `mcr.microsoft.com/dotnet/sdk:6.0` tag, an earlier version may be cached on your machine.  To ensure the latest use the `--pull` option when building and/or running to ensure the latest.  Be sure to read the documentation because the option behaves differently between the various commands.

**Comment by MikeHingley22** [comment]
I'll give it a go - this is from a GitHub Action so I'll add that reporting in.

**Comment by MikeHingley22** [comment]
@MichaelSimons  - I just tried it locally - and it does indeed pull the version that should have this fixed :

```
#11 [build  3/13] RUN dotnet --info
#11 0.640 .NET SDK (reflecting any global.json):
#11 0.640  Version:   6.0.416
#11 0.640  Commit:    db3bc4a6c6
#11 0.640
#11 0.640 Runtime Environment:
#11 0.643  OS Name:     debian
#11 0.643  OS Version:  11
#11 0.644  OS Platform: Linux
#11 0.666  RID:         debian.11-x64
#11 0.666  Base Path:   /usr/share/dotnet/sdk/6.0.416/
#11 0.667
#11 0.667 global.json file:
#11 0.667   Not found
#11 0.667
#11 0.667 Host:
#11 0.667   Version:      6.0.24
#11 0.667   Architecture: x64
#11 0.667   Commit:       e7b8488daf
#11 0.667
#11 0.667 .NET SDKs installed:
#11 0.667   6.0.416 [/usr/share/dotnet/sdk]
#11 0.667
#11 0.667 .NET runtimes installed:
#11 0.667   Microsoft.AspNetCore.App 6.0.24 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
#11 0.667   Microsoft.NETCore.App 6.0.24 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
#11 0.667
#11 0.667 Download .NET:
#11 0.667   https://aka.ms/dotnet-download
#11 0.667
#11 0.667 Learn about .NET Runtimes and SDKs:
#11 0.667   https://aka.ms/dotnet/runtimes-sdk-info
#11 DONE 0.8s
```

I think I made the assumption that Trivy Container scanning was... scanning the container.  I'm going to check this container I just made to see if it still shows that CVE.
